File: gdx2d-desktop/src/hevs/gdx2d/demos/complex_shapes/DrawableShape.java
package hevs.gdx2d.demos.complex_shapes;

import java.util.Random;

import com.badlogic.gdx.graphics.Color;

/**
 * A drawable shape class for the {@link DemoComplexShapes} demonstration
 * 
 * @author Pierre-Andre Mudry (mui)
 * @version 1.0
 */
public class DrawableShape {

	private static Random rnd = new Random();
	
	int width, height;
	int x, y;
	float offset;
	Color c;

	DrawableShape(int w, int h, int x, int y, Color c) {
		this.width = w;
		this.height = h;
		this.x = x;
		this.y = y;
		this.c = c;
		this.offset = rnd.nextFloat() * 45;
	}
}

File: gdx2d-demoDesktop/src_to_do/main/kotlin/ch/hevs/gdx2d/demos/complex_shapes/DrawableShape.kt
package ch.hevs.gdx2d.demos.complex_shapes

import com.badlogic.gdx.graphics.Color

import java.util.Random

/**
 * A drawable shape class for the [DemoComplexShapes] demonstration
 *
 * @author Pierre-Andre Mudry (mui)
 * @version 1.0
 */
class DrawableShape internal constructor(var width: Int, var height: Int, var x: Int, var y: Int, var c: Color) {
    var offset: Float = 0.toFloat()

    init {
        this.offset = rnd.nextFloat() * 45
    }

    companion object {

        private val rnd = Random()
    }
}
